variables.tf

```diff
@@ -137,12 +143,27 @@ variable "site_config" {
   default = {}
 }
 
-variable "identity_ids" {
-  description = <<EOT
-    Specifies a list of user managed identity ids to be assigned.
-  EOT
-  type        = list(string)
+variable "identity" {
+  description = "(Optional) An identity block."
+  type = object({
+    type         = string
+    identity_ids = optional(list(string))
+  })
+  default = null
+  validation {
+    condition     = var.identity == null || can(contains(["SystemAssigned", "UserAssigned", "SystemAssigned, UserAssigned"], var.identity.type))
```

Comment:
just FYI. I have had problems using logical operators like || && in validation as they are not short circuit. So when identity== null, the 2nd condition would fail. but looks like the `can()` is making it to work https://developer.hashicorp.com/terraform/language/expressions/operators

Reply:
I have run into this as well. It is working in this case and I have accounted for both the null condition and the non-null condition in tests.